modules/src/core/ics04_channel/handler/acknowledgement.rs

```diff
@@ -19,10 +22,15 @@ pub struct AckPacketResult {
     pub seq_number: Option<Sequence>,
 }
 
-pub fn process(
-    ctx: &dyn ChannelReader,
+pub(crate) fn process<Ctx, Cap>(
```

Comment:
All ICS04 handlers are now generic over `Capability` and authenticate the provided capability before continuing with actual handler logic.